### PR TITLE
Fix organelle unlock conditions displaying incorrect amounts and units

### PR DESCRIPTION
**Pull request by Oliveriver**
**Brief Description of What This PR Does**

Formats the compound amount in unlock conditions to match the format of the same compound in the patch details panel.

**Related Issues**

N/A

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [ ] Initial code review passed (this and further items should not be checked by the PR author)
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: src/microbe_stage/organelle_unlocks/WorldBasedUnlockCondition.cs [hunk 1]
```diff
@@ -144,17 +146,46 @@ public override void GenerateTooltip(LocalizedStringBuilder builder, IUnlockStat
         {
             var compoundName = Compound!.InternalName;
 
-            if (Min.HasValue && Max.HasValue)
+            var percentageFormat = TranslationServer.Translate("PERCENTAGE_VALUE");
```
Comment (hhyyrylainen) [inline src/microbe_stage/organelle_unlocks/WorldBasedUnlockCondition.cs:149]: Considering how this is meant to generate text that can actually support being re-localized, these formattings shouldn't be applied, but rather these need to be created as more LocalizedString instances with these keys and then the values.

File: src/microbe_stage/organelle_unlocks/WorldBasedUnlockCondition.cs [hunk 2]
```diff
@@ -144,17 +146,46 @@ public override void GenerateTooltip(LocalizedStringBuilder builder, IUnlockStat
...
+            var unitFormat = TranslationServer.Translate("VALUE_WITH_UNIT");
+
+            string? formattedMin;
+            string? formattedMax;
+
+            switch (compoundName)
+            {
+                case "sunlight":
+                    formattedMin = Min.HasValue ? unitFormat.FormatSafe(percentageFormat.FormatSafe(Min), "lx") : null;
```
Comment (hhyyrylainen) [inline src/microbe_stage/organelle_unlocks/WorldBasedUnlockCondition.cs:158]:
```suggestion
                    formattedMin = Min.HasValue ? new LocalizedString("PERCENTAGE_VALUE", new LocalizedString("PERCENTAGE_VALUE", Min)), "lx") : null;
```
Something like that.

**Review by ARecurringProblem — APPROVED** [review]

File: src/microbe_stage/organelle_unlocks/WorldBasedUnlockCondition.cs (hunk 2 above)
Comment (hhyyrylainen) [reply src/microbe_stage/organelle_unlocks/WorldBasedUnlockCondition.cs:158]: I'm not sure I fully understand, but storing a LocalizedString in a temporary variable might make the design easier to do?

**Comment by hhyyrylainen** [comment]
I'll try to fix the code localization object use myself to get this done now to have this ready well in advance of the release.

**Review by hhyyrylainen — APPROVED** [review]
I updated this to keep stuff as LocalizedString instances to keep the re-translatability support. I quickly playtested this, but some double checking for my changes would be good.

**Pull request merged**